similar/main/gamecntl.cpp

```diff
@@ -1304,7 +1304,10 @@ static window_event_result HandleTestKey(int key)
 		case KEY_E + KEY_DEBUGGED:
 			window_set_visible(Game_wind, 0);	// don't let the game do anything while we set the editor up
 			init_editor();
-			return window_event_result::close;
+			// If editor failed to load, carry on playing
+			if (!EditorWindow)
+				window_set_visible(Game_wind, 1);
+			return EditorWindow ? window_event_result::close : window_event_result::handled;
```

Comment:
The compiler will likely assume that `window_set_visible` could modify `EditorWindow`.  You could improve the generated code by leaving the original `return` as-is and writing the recovery path as:

```c++
    if (!EditorWindow)
    {
        window_set_visible(Game_wind, 1);
        return window_event_result::handled;
    }
```

This would discourage the compiler from rereading `EditorWindow` after `window_set_visible` returns.